format-common/src/test/java/io/cdap/plugin/format/BigDecimalAwareStructuredRecordStringConverterTest.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.format;
+
+import io.cdap.cdap.api.data.format.StructuredRecord;
+import io.cdap.cdap.api.data.schema.Schema;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.math.MathContext;
+
+public class BigDecimalAwareStructuredRecordStringConverterTest {
+  private static final Schema SCHEMA = Schema.recordOf(
+    "rec",
+    Schema.Field.of("i", Schema.of(Schema.Type.INT)),
+    Schema.Field.of("f", Schema.of(Schema.Type.FLOAT)),
+    Schema.Field.of("bd", Schema.decimalOf(32, 7)),
+    Schema.Field.of("d", Schema.of(Schema.Type.DOUBLE)),
+    Schema.Field.of("l", Schema.of(Schema.Type.LONG)));
+
+  private static final BigDecimal BD =
+    new BigDecimal("12398321.8127312", MathContext.DECIMAL128).setScale(7);
+
+  private static final StructuredRecord RECORD = StructuredRecord.builder(SCHEMA)
+    .set("i", 1)
+    .set("f", 256.1f)
+    .setDecimal("bd", BD)
```

Comment:
Please also test for `.setDecimal(.., null)

Reply:
Added